pkg/logging/sftp/create.go

```diff
@@ -0,0 +1,154 @@
+package sftp
+
+import (
+	"io"
+
+	"github.com/fastly/cli/pkg/common"
+	"github.com/fastly/cli/pkg/compute/manifest"
+	"github.com/fastly/cli/pkg/config"
+	"github.com/fastly/cli/pkg/errors"
+	"github.com/fastly/cli/pkg/text"
+	"github.com/fastly/go-fastly/fastly"
+)
+
+// CreateCommand calls the Fastly API to create SFTP logging endpoints.
+type CreateCommand struct {
+	common.Base
+	manifest manifest.Data
+
+	// required
+	EndpointName  string // Can't shaddow common.Base method Name().
+	Version       int
+	Address       string
+	User          string
+	SSHKnownHosts string
+
+	// optional
+	Port              common.OptionalUint
+	Password          common.OptionalString
+	PublicKey         common.OptionalString
+	SecretKey         common.OptionalString
+	Path              common.OptionalString
+	Period            common.OptionalUint
+	Format            common.OptionalString
+	FormatVersion     common.OptionalUint
+	GzipLevel         common.OptionalUint
+	ResponseCondition common.OptionalString
+	TimestampFormat   common.OptionalString
+	Placement         common.OptionalString
+}
+
+// NewCreateCommand returns a usable command registered under the parent.
+func NewCreateCommand(parent common.Registerer, globals *config.Data) *CreateCommand {
+	var c CreateCommand
+
+	c.Globals = globals
+	c.manifest.File.Read(manifest.Filename)
+	c.CmdClause = parent.Command("create", "Create an SFTP logging endpoint on a Fastly service version").Alias("add")
+
+	c.CmdClause.Flag("name", "The name of the SFTP logging object. Used as a primary key for API access").Short('n').Required().StringVar(&c.EndpointName)
+	c.CmdClause.Flag("service-id", "Service ID").Short('s').StringVar(&c.manifest.Flag.ServiceID)
+	c.CmdClause.Flag("version", "Number of service version").Required().IntVar(&c.Version)
+
+	c.CmdClause.Flag("address", "The hostname or IPv4 addres").Required().StringVar(&c.Address)
+	c.CmdClause.Flag("user", "The username for the server").Required().StringVar(&c.User)
+	c.CmdClause.Flag("ssh-known-hosts", "A list of host keys for all hosts we can connect to over SFTP").Required().StringVar(&c.SSHKnownHosts)
+
+	c.CmdClause.Flag("port", "The port number").Action(c.Port.Set).UintVar(&c.Port.Value)
```

Comment:
@ezkl good catch, so this is actually required. I was considering making this required, but thought that defaults were set in the API because I saw values populated in the UI of manage.fastly.com.

Reply:
We discussed offline. This field should actually receive a reasonable default downstream and therefore, not be required by the CLI API.

The error message is reasonable enough the we are okay merging this.